### PR TITLE
Database preparation scripts

### LINKED ISSUE #6: DB Schema Design

### PR DESCRIPTION
**Pull request by rafraser**
This pull request contains two key SQL files and some example queries. For more information about the database structure, see the [Database Structure wiki page](https://github.com/summerprogram2019/ecosoft/wiki/Database-Structure)

- create.sql creates the required tables
- data.sql has some example testing data
- Sample queries for leaderboards and other useful functionality

Closes #6

### CODE REVIEW

File: db/create.sql [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  `uid` int(11) NOT NULL AUTO_INCREMENT,
+  `username` tinytext NOT NULL,
+  `password` tinytext NOT NULL,
+  PRIMARY KEY (`uid`),
+  UNIQUE KEY `Username Unique` (`username`(16))
+) COMMENT='Stores the basic user identification settings such as username and password. These are primarily used for login.';
+
+CREATE TABLE IF NOT EXISTS `activities` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `name` tinytext NOT NULL,
```
Comment (Guthers) [inline db/create.sql]: Should use varchar and specify size

File: db/create.sql [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  `price` int(11) NOT NULL,
+  `instore` enum('T','F') NOT NULL,
+  PRIMARY KEY (`id`)
+) COMMENT='Keeps track of the different flower types';
+
+CREATE TABLE IF NOT EXISTS `friends` (
+  `user1` int(11) NOT NULL,
+  `user2` int(11) NOT NULL,
+  PRIMARY KEY (`user1`,`user2`),
+  KEY `FK_friends_user_2` (`user2`),
```
Comment (Guthers) [inline db/create.sql:41]: Should there not also be a user_1 key?

File: db/create.sql [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  `planted` datetime NOT NULL,
+  PRIMARY KEY (`uid`,`flower`,`planted`),
+  KEY `FK_garden_flowers` (`flower`),
+  CONSTRAINT `FK_garden_flowers` FOREIGN KEY (`flower`) REFERENCES `flowers` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `FK_garden_user` FOREIGN KEY (`uid`) REFERENCES `user` (`uid`) ON DELETE CASCADE ON UPDATE CASCADE
+) COMMENT='Keeps track of all the flowers that users have in their gardens';
+
+CREATE TABLE IF NOT EXISTS `goals` (
+  `uid` int(11) NOT NULL,
+  `description` text NOT NULL,
```
Comment (Guthers) [inline db/create.sql]: again varchar?

File: db/create.sql [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  `title` text NOT NULL,
+  `description` text,
+  `url` text NOT NULL,
+  `date` date NOT NULL,
+  PRIMARY KEY (`id`)
+) COMMENT='Stores information about the articles that may appear in user feeds.';
+
+CREATE TABLE IF NOT EXISTS `flowers` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `name` tinytext NOT NULL,
```
Comment (Guthers) [inline db/create.sql]: varchar?

File: db/create.sql [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  `points` smallint(6) NOT NULL,
+  PRIMARY KEY (`id`)
+) COMMENT='Keeps track of the various activities that a user can perform to earn points';
+
+CREATE TABLE IF NOT EXISTS `articles` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `title` text NOT NULL,
+  `description` text,
```
Comment (Guthers) [inline db/create.sql:22]: varchar?

File: db/create.sql [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+CREATE TABLE IF NOT EXISTS `activities` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `name` tinytext NOT NULL,
+  `points` smallint(6) NOT NULL,
+  PRIMARY KEY (`id`)
+) COMMENT='Keeps track of the various activities that a user can perform to earn points';
+
+CREATE TABLE IF NOT EXISTS `articles` (
+  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `title` text NOT NULL,
```
Comment (Guthers) [inline db/create.sql]: varchar?

File: db/data.sql [hunk 7]
```diff
@@ -0,0 +1,92 @@
+INSERT INTO `user` (`uid`, `username`, `password`) VALUES
+	(1, 'rafraser', 'klagjaihsakg'),
```
Comment (Guthers) [inline db/data.sql:2]: Are these the plain text passwords? If not, what has being done to them?

File: db/sample_queries/articlelikes.sql [hunk 8]
```diff
@@ -0,0 +1,4 @@
+SELECT a.title AS 'Article Name', COUNT(l.uid) AS 'Number Of Likes'
```
Comment (Guthers) [inline db/sample_queries/articlelikes.sql:2]: Should add a commented out line at the start describing what the sample queries do

**Review by Guthers — CHANGES_REQUESTED** [review]

File: db/data.sql (hunk 7 above)
Comment (rafraser) [reply db/data.sql:2]: Discussion in #35

File: db/sample_queries/articlelikes.sql (hunk 8 above)
Comment (rafraser) [reply db/sample_queries/articlelikes.sql:2]: Resolved in 4fb0fe2

File: db/create.sql (hunk 1 above)
Comment (rafraser) [reply db/create.sql]: Resolved in 864b5d0

File: db/create.sql (hunk 3 above)
Comment (rafraser) [reply db/create.sql]: Resolved in 864b5d0

File: db/create.sql (hunk 4 above)
Comment (rafraser) [reply db/create.sql]: Resolved in 864b5d0

File: db/create.sql (hunk 5 above)
Comment (rafraser) [reply db/create.sql:22]: Resolved in 864b5d0

File: db/create.sql (hunk 6 above)
Comment (rafraser) [reply db/create.sql]: Resolved in 864b5d0

File: db/create.sql (hunk 2 above)
Comment (rafraser) [reply db/create.sql:41]: Resolved in 1b50d70

**Review by Guthers — APPROVED** [review]

**Pull request merged**